src/librustdoc/html/static/js/main.js

```diff
@@ -888,7 +888,7 @@ function hideThemeButtonState() {
 
     onEachLazy(document.querySelectorAll(".rustdoc-toggle > summary:not(.hideme)"), function(el) {
         el.addEventListener("click", function(e) {
-            if (e.target.tagName != "SUMMARY") {
+            if (e.target.tagName != "SUMMARY" && e.target.tagName != "A") {
```

Comment:
This is definitely an improvement, but it seems somewhat fragile. Are there other tags that are used as click targets?

Reply:
Buttons and forms have a default behavior on click, but those don't show up in our summaries. You're right that this feels fragile, though. Perhaps as a followup we should flip the logic and say "preventDefault if the click target is `h3`, `h4`, etc". That seems to be the usual click target in these when it's not the anchor, but need to do a little testing to confirm.

Reply:
A quick check shows we'd also have to cover `<div class="stab unstable">` and friends - the item-info divs.

Reply:
> but those don't show up in our summaries.

> A quick check shows we'd also have to cover `<div class="stab unstable">` and friends - the item-info divs.

I thought item-infos don't show up in the summaries either? Or do they and I'm misunderstanding?

Reply:
item-infos are currently inside the summary. The idea is that they should be shown even when the whole document is collapsed, so you can see at a glance what is deprecated, unstable, platform-specific, etc.

Those and the "notable traits" hover div are the main surprising things that are inside summaries right now.